contextualized/regression/datamodules.py

```diff
@@ -0,0 +1,133 @@
+            w (ndarray): 3D array containing known 2D network per each sample.
+            dataset (str): Which dataset to use. Choose between ["multivariate", "univariate", "multitask_multivariate", "multitask_univariate].
+            n (int): Number of data samples to use. Defaults to 0 (full dataset will be used).
+            correlation (bool): Whether datamodule will be used for correlation regression module.
+            markov (bool): Whether datamodule will be used for markov regression module. (Currently unused)
+            num_workers (int): Number of workers used in dataloaders.
+            batch_size (int): Size of batches used in dataloaders.           
+        """
+
+        super().__init__()
+
+        self.dataset = DATASETS[dataset]
+        self.num_workers = 0
+        self.batch_size = batch_size
+        if n == 0:
+            n = c.shape[0]  # use full dataset
+
+        self.n = n
+        self.C = torch.tensor(c)
+        self.X = torch.tensor(x)
+        self.Y = torch.tensor(y)
+
+        if correlation or markov:
+            self.Y = self.X
+
+        # partition data
+        self.train_idx, self.test_idx, self.val_idx = self._create_idx(self.n)
+        self.full_dataset = self.dataset(self.C, self.X, self.Y, idx=range(self.n - 80))
+        self.train_dataset = self.dataset(self.C, self.X, self.Y, idx=self.train_idx)
+        self.test_dataset = self.dataset(self.C, self.X, self.Y, idx=self.test_idx)
+        self.val_dataset = self.dataset(self.C, self.X, self.Y, idx=self.val_idx)
+        # self.pred_dataset = self.dataset(self.C, self.X, self.Y, idx=self.pred_idx)
+
+    def setup(self, stage: str):
+
+        # Assign test/train/val datasets for use in dataloaders
+        if stage == "fit":
+            pass
+            # print('fit running')
+            # self.train_dataset = test_dataset_11(self.C, self.X, self.Y, idx=self.train_idx) #self.train_idx)
+            # self.C_train, self.X_train, self.Y_train = self.train_dataset.unpack()
+
+        elif stage == "test" or stage == "predict":
+            pass
+            # print('predicting')
+            # self.test_dataset = test_dataset_11(self.C, self.X, self.Y, idx=self.test_idx)
+            # self.C_test, self.X_test, self.Y_test = self.test_dataset.unpack()
+
+        elif stage == "validate":
+            pass
+            # print('validating')
+            # self.C_val, self.X_val, self.Y_val = self.val_dataset.unpack()
+
+    def train_dataloader(self):
+        return DataLoader(
+            self.train_dataset,
+            batch_size=self.batch_size,
+            shuffle=True,
+            num_workers=self.num_workers,
+        )
+
+    def test_dataloader(self):
+        return DataLoader(
+            self.test_dataset,
+            batch_size=self.batch_size,
+            shuffle=False,
+            num_workers=self.num_workers,
+        )
+
+    def predict_dataloader(self):
```

Comment:
Similar to above, we'll want to predict for train, test, and val. This seems like `trainer.predict` will only return val predictions.